### ISSUE TITLE
Фильтры и Хуки в 1.4

### ISSUE THREAD

**Issue by me-2-you** [kalbac/wc-edostavka#26]
А можно пример применения новшеств из 1.4 =) Хочу сменить имена тарифов (как в прошлой версии Курьерская доставка и ПВЗ), но что-то делаю явно не так, не выходит =(

**Comment by kalbac** [comment]
Для изменения названия тарифов используйте фильтр edostavka_delivery_tariff_name
Пример:
```
add_filter( 'edostavka_delivery_tariff_name', 'custom_edostavka_delivery_tariff_name', 10, 2 );
function custom_edostavka_delivery_tariff_name( $label, $tariff_id ) {
 if( 136 == absint( $tariff_id ) ) {
  $label = 'Моё название тарифа';
 }
 return $label;
}
```

**Comment by kalbac** [comment]
Фильтр **edostavka_delivery_tariffs** содержит массив тарифов в виде id => (numeric) label => название тарифа и type => тип тарифа (до двери [door]| до склада [stock])
Тарифы до постоматов уже не актуальны.

**Пример добавления нового тарифа**:
```
add_filter( 'edostavka_delivery_tariffs', 'add_my_custom_tariff');
function add_my_custom_tariff( $tariffs ) {
 array_push( $tariffs, array( 'id' => 158, 'label' => 'Международный экспресс', 'type' => 'stock'));
 return $tariffs;
}
```

**Пример удаления тарифа**:
```
add_filter( 'edostavka_delivery_tariffs', 'remove_tariff');
function remove_tariff( $tariffs ) {
 foreach( $tariffs as $index => $tariff ) {
  if( 158 == $tariff['id'] ) {
   unset( $tariffs[$index] );
  }
 }
 return $tariffs;
}
```

**Comment by alexey13** [comment]
`add_filter( 'edostavka_delivery_tariff_name', 'custom_edostavka_delivery_tariff_name', 10, 2 );
function custom_edostavka_delivery_tariff_name( $label, $tariff_id ) {
 if( 136 == absint( $tariff_id ) {
  $label = 'Моё название тарифа';
 }
 return $label;
}`

При использовании этого кода получаю ошибку:
`syntax error, unexpected '{'
in line:
if( 136 == absint( $tariff_id ) {
  $label = 'СДЭК доставка до пункта выдачи';
 }
 return $label;
}`

**Comment by alexey13** [comment]
В коде скобка пропущена

**Comment by kalbac** [comment]
Пример кода писался "на коленке". Спасибо что нашли опечатку.

**Comment by me-2-you** [comment]
Хмм, а если названия менять у нескольких тарифов? Как возвращать названия в одной функции? Это можно вообще или нет?

**Comment by kalbac** [comment]
@me-2-you Конечно... Делайте либо через switch либо просто для каждого тарифа отдельный if
`if( 136 == $tariff_id ) //do rename tariff`

**Comment by me-2-you** [comment]
@kalbac так я по этому и спрашивал... потому что когда я делаю два if или через switch, то начинает выводится только один переписанный метод с нужным названием, другие методы тупо исчезают... 

Хотя сейчас вообще ничего не работает =( Все поудалял, но даже стандартные методы доставки не показываются...

**Comment by kalbac** [comment]
@me-2-you 
```
if( 136 == absint( $tariff_id ) ) {
  $label = 'Моё название тарифа';
 } elseif( 135 == absint( $tariff_id ) ) {
  $label = 'Другое название тарифа';
 }
```

**Comment by kalbac** [comment]
Щас пришла ещё одна мысль, как переименовать несколько тарифов не применяя кучу **IF** 

```
add_filter( 'edostavka_delivery_tariff_name', 'custom_edostavka_delivery_tariff_name', 10, 2 );
function custom_edostavka_delivery_tariff_name( $label, $tariff_id ) {
 $my_custom_names = array( 136 => 'Первое название', 135 => 'Второе название' ); //В общем создаём массив с вашими названиями тарифов.
 if( in_array( $tariff_id, array_keys( $my_custom_names ) ) ) {
  $label = $my_custom_names[$tariff_id];
 }
/*
* Или ещё проще $label = isset( $my_custom_names[$tariff_id] ) ? $my_custom_names[$tariff_id] : $label;
*/
 return $label;
}
```

**Comment by dimsonbgg** [comment]
А если сделать еще проще, переименовать название тарифа в файле funсtion.php самого плагина?

function wc_edostavka_get_delivery_tariffs() {
	return apply_filters( 'edostavka_delivery_tariffs', array(
		array(
                    .............
		array(
			'id'	=> 10,
			'label'	=> 'Доставка до пункта выдачи заказов "СДЭК"',
			'type'	=> 'stock'
		),
		array(
			'id'	=> 11,
			'label'	=> 'Доставка курьером "СДЭК" до двери',
			'type'	=> 'door'
		),
.............

**Comment by kalbac** [comment]
@dimsonbgg Это не будет проще, потому что при следующем обновлении у вас всё "слетит"

**Comment by fishfoto** [comment]
Могу вас попросить написать правленный фильтр, а то как не пробую все время сайт укладываю.
Найти ошибку не в силах, не понимаю синтаксиса. 
Спасибо.

**Comment by dimsonbgg** [comment]
Максим вам больше поможет. А так, опишите проблему по подробнее, может найдем общее решение проблемы, укажите ссылку на сайт(страницу).

**Comment by sergeygubanoff** [comment]
> Для изменения названия тарифов используйте фильтр edostavka_delivery_tariff_name Пример:

Подскажите пожалуйста, куда и в какой файл вставить этот кусок кода, для изменения названия доставки?

**Comment by kalbac** [comment]
@sergeygubanoff В файл functions.php в вашей активной теме. Но проще всего будет просто переименовать нужные тарифы в файле functions.php в плагине.
